### DISCUSSION TITLE
Intl.DisplayNames equivalent?

### DISCUSSION THREAD

**Discussion by hydroper** [airsdk/Adobe-Runtime-Support#2521] [Ideas]
It looks like Flash ships with Unicode CLDR data, so I suppose it should make available... something like Ecma-262 [Intl.DisplayNames](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames)?

For example, [flash.globalization.DateTimeFormatter](https://airsdk.dev/reference/actionscript/3.0/flash/globalization/DateTimeFormatter.html) formats date-time just like Ecma-262's `Intl.DateTimeFormat`. I saw the ICU4X project (for Rust) on GitHub and saw they require a data provider.

If Flash/AIR ships with Unicode data, does it support `DisplayNames`? It's basically data for mapping from a locale identifier to a proper language or region name. If it ships with, but doesn't provide it, then it means we've to duplicate data. Or doesn't it ship with such data?

**Comment by ajwfrost** [comment]
Hi

We don't have any built-in support for this; it should be fairly straightforward to create this based on the data available at https://cldr.unicode.org/ with some XML-based logic to create an API like you reference there.. 

There shouldn't be (much) duplication; internally for this sort of capability currently (like datetime formatting) there is a library that Adobe developed themselves many years ago which uses platform-specific implementations to hook into the relevant OS function calls for these capabilities..

thanks
